### PR TITLE
Use associative array to register mapping

### PR DESCRIPTION
**Pull request by Toilal**
This should enhance performance of AutoMapperConfig hasMappingFor and getMappingFor methods.

### CODE REVIEW

File: src/Configuration/AutoMapperConfig.php [hunk 1]
```diff
@@ -55,12 +55,9 @@ public function getMappingFor(
         string $destinationClassName
     ): ?MappingInterface {
         // Check for an exact match before we try parent classes.
-        foreach ($this->mappings as $mapping) {
-            $isExactMatch = $mapping->getSourceClassName() === $sourceClassName
-                && $mapping->getDestinationClassName() === $destinationClassName;
-            if ($isExactMatch) {
-                return $mapping;
-            }
+        $mapping = @$this->mappings["$sourceClassName|$destinationClassName"];
```
Comment (BoShurik) [inline src/Configuration/AutoMapperConfig.php]: `$mapping = $this->mappings["$sourceClassName|$destinationClassName"] ?? null` is faster
```
$test = [];
$start = microtime(true);
for ($i = 0; $i < 10000000; $i++) {
    $result = @$test[$i];
}

echo (microtime(true) - $start) . \PHP_EOL; // 10.039760112762
```
vs
```
$test = [];
$start = microtime(true);
for ($i = 0; $i < 10000000; $i++) {
    $result = $test[$i] ?? null;
}

echo (microtime(true) - $start) . \PHP_EOL; // 2.923015832901
```

**Review by BoShurik — CHANGES_REQUESTED** [review]

File: src/Configuration/AutoMapperConfig.php (hunk 1 above)
Comment (Toilal) [reply src/Configuration/AutoMapperConfig.php]: I was not aware of this operator, thanks.

Comment (Toilal) [reply src/Configuration/AutoMapperConfig.php]: But it was added in PHP 7.0, do I need to support 5.6 and use isset with ternary operator instead?

Comment (BoShurik) [reply src/Configuration/AutoMapperConfig.php]: Library did not support it either: https://github.com/mark-gerarts/automapper-plus/blob/master/composer.json#L24

Comment (andrew-demb) [reply src/Configuration/AutoMapperConfig.php]: This library requires minimal php version as 7.1

**Comment by mark-gerarts** [comment]
Definitely a performance improvement, and cleaner code on top of that!

**Pull request merged**